### ISSUE TITLE
NVDA and last version of BRLTTY

### ISSUE THREAD

**Issue by pzajda** [nvaccess/nvda#6764]
Hi,

I use Windows 7 64-bits.
And NVDA appears to be not compatible with last BRLTTY (5.4 at this time).
When NVDA starts connecting to BRLTTY, BRLTTY service crashes.
So I tested all 5.X releases of BRLTTY for Windows:

- Version 5.0: looks like OK, for a fresh or an upgrade from version 4.5;
- Version 5.1: OK, i use it while creating this issue;
- Version 5.2: at the beginning I though it was OK, but sometimes the braille display is not refreshed, sometimes it is but after a some seconds latency;
- Version 5.3, 5.3.1 and 5.4: doesn't work at all. BRLTTY service crashes as soon as NVDA begins to use it.
The NVDA log mentions the connection to BRLTTY was not possible.

I also wrote to BRLTTY discussion list, unfortunately without finding a solution.
And it really seems BRLTTY crashes at the moment NVDA initialize it.
While NVDA doesn't use it, BRLTTY service is running correctly.

Maybe a BRLAPI version compatibility issue, as you answered @JCSTeh on nvda-devel discussion list this week.

**Comment by josephsl** [comment]
Cc @JCSTeh

**Comment by jcsteh** [comment]
P1 because we're crashing BRLTTY.

**Comment by pzajda** [comment]
I have just had the following answer from Dave Mielke on BRLTTY discussion list.

> I know that nvda uses the Python bindings for BrlAPI. It may have been around then that we became compatible with Python 3.

He mentioned that after I specified BRLTTY began to crash since version 5.3.

**Comment by jcsteh** [comment]
It looks like BRLTTY now only distributes Windows Python bindings for Python 3.4. We need bindings for Python 2.7. @pzajda, would you mind asking if someone has recent BRLAPI bindings for Python 2.7?

In the short term, I think we should just recommend that users use 5.1, since building BRLTTY ourselves is not something we have time to investigate right now.

**Comment by pzajda** [comment]
Hi,

Unfortunately, I haven't had any answer when asking if a 2.7 version of BRLAPI Python binding exists somewhere and Google doesn't seem to have more answer.
I sent my mail on January 29th so I don't think I am inpatient if I suppose I won't have any answer latter.

Best regards.

**Comment by gregjozk** [comment]
Hi,

I have tested last version of BRLTTY (released today), on my computer and it works with NVDA so far. hope, it will work tomorrow too.

here is also a request from BRLTTY developers:

Both the NVDA fix and the MS update to libusb can be tested via the
brltty-win-5.4-233 files that are now available on brltty's download page.

Please test soon as 5.5 will most likely be released in a little over a week
from now.

regards, Jožef

**Comment by pzajda** [comment]
@gregjozk This version also works for me, it is a really good new.
Finally the issue was not in the Python binding but in BRLTTY itself.
This is a test release before BRLTTY 5.5. So the next final version of BRLTTY (5.5) should work with NVDA.
So I was a little bit impatient.

**Comment by jcsteh** [comment]
BRLTTY 5.5 has now been released. Closing.
